src/main/kotlin/com/pravo/pravo/domain/promise/repository/PromiseRepositoryImpl.java

```diff
@@ -24,32 +25,35 @@ public PromiseRepositoryImpl(EntityManager entityManager) {
     @Override
     public List<PromiseResponseDto> getPromisesByMemberIdAndStartedAtAndEndedAt(Long memberId,
         LocalDate startedAt, LocalDate endedAt) {
+        QPromiseRole promiseRole = QPromiseRole.promiseRole;  // 기본 alias
+        QPromiseRole organizer = new QPromiseRole("organizer");  // 모임장용 alias
+        QPromise promise = QPromise.promise;
+
         return queryFactory
             .select(new QPromiseResponseDto(
                 promise.id,
                 promise.name,
                 promise.promiseDate,
                 promise.location,
                 promise.status,
-                promiseRole.member.name,
-                promiseRole.member.profileImageUrl
+                organizer.member.name,
+                organizer.member.profileImageUrl
             ))
-            .from(promiseRole)
-            .join(promiseRole.promise, promise)
-            .leftJoin(promiseRole.member)
+            .from(promise)
+            .join(promiseRole).on(promiseRole.promise.eq(promise))
+            .leftJoin(organizer).on(
+                organizer.promise.eq(promise)
+                    .and(organizer.role.eq(RoleStatus.ORGANIZER))
+            )
```

Comment:
```suggestion
            .leftJoin(QPromiseRole.promiseRole, "organizer").on(
                organizer.promise.eq(promise)
                    .and(organizer.role.eq(RoleStatus.ORGANIZER))
            )
```
P3: 이런식으로 선언하면 위에 Q타입 객체를 따로 선언 안하고 바로 이용 가능할 것 같은데 어떠신가요?_?

Reply:
오호 별거 아닌데 더 간단한 것 같습니다 수정해서 머지하겠씁니다~

Reply:
오 이렇게 수정하면 위에서 ``` .select(new QPromiseResponseDto(
            promise.id,
            promise.name,
            promise.promiseDate,
            promise.location,
            promise.status,
            organizer.member.name,
            organizer.member.profileImageUrl
        ))``` organizer를 쓰지 못하네요  😅 어쩔수 없이 기존 방식대로 해야할 것 같습니다